typing/ctype.ml

```diff
@@ -2514,7 +2516,10 @@ let complete_type_list ?(allow_absent=false) env nl1 lv2 mty2 nl2 tl2 =
         match Env.find_type_by_name lid env' with
         | (_, {type_arity = 0; type_kind = Type_abstract;
                type_private = Public; type_manifest = Some t2}) ->
-            (n, nondep_instance env' lv2 id2 t2) :: complete nl ntl2
+            begin match nondep_instance env' lv2 id2 t2 with
+            | exception Nondep_cannot_erase _ -> raise Exit
```

Comment:
Looking at the 4.07 code, a failure of `nondep_instance` was equivalent to `Exit` only in the case `allow_absent=false`. What is the reason for this change?

Reply:
> What is the reason for this change?

My inattention.
Thanks for the catch.